lib/Echidna/Solidity.hs

```diff
@@ -22,37 +23,46 @@ import EVM
   (VM, VMResult(..), contract, gas, loadContract, replaceCodeOfSelf, resetState, state)
 import EVM.Concrete (Blob(..))
 import EVM.Exec     (exec, vmForEthrunCreation)
-import EVM.Solidity (abiMap, contractName, creationCode, methodInputs, methodName, readSolc)
+import EVM.Solidity (abiMap, contractName, creationCode, methodInputs, methodName, readSolc, SolcContract)
 
 data EchidnaException = CompileFailure
                       | NoContracts
                       | TestArgsFound Text
+                      | ContractNotFound Text
 
 instance Show EchidnaException where
-  show CompileFailure    = "Couldn't compile given file"
-  show NoContracts       = "No contracts found in given file"
-  show (TestArgsFound t) = "Test " ++ show t ++ " has arguments, aborting"
+  show CompileFailure       = "Couldn't compile given file"
+  show NoContracts          = "No contracts found in given file"
+  show (ContractNotFound c) = "Given contract " ++ show c ++ " not found in given file"
+  show (TestArgsFound t)    = "Test " ++ show t ++ " has arguments, aborting"
 
 instance Exception EchidnaException
 
-loadSolidity :: (MonadIO m, MonadThrow m) => FilePath -> m (VM, [SolSignature], [Text])
-loadSolidity f = liftIO solc >>= \case
+selectContract :: (MonadThrow m) => [SolcContract] -> Maybe Text -> m SolcContract
+selectContract [] _ = throwM NoContracts
+selectContract (c:_) Nothing = return c
+selectContract cs (Just name) = case find (\x -> name == x ^. contractName) cs of
+  Nothing -> throwM $ ContractNotFound name
+  Just c  -> return c
+
+loadSolidity :: (MonadIO m, MonadThrow m) => FilePath -> Maybe Text -> m (VM, [SolSignature], [Text])
```

Comment:
Can we refactor the example in README.md to use this new convention?

Reply:
sure, will do